pkg/wallet/wallet.go

```diff
@@ -86,6 +95,186 @@ func (w *BaseWallet) CreateSigningKey() (string, error) {
 	return base58Pub, nil
 }
 
+// FindVerKey selects a verification key which is present in candidateKeys that is present in the wallet
+func (w *BaseWallet) FindVerKey(candidateKeys []string) (int, error) {
+	for i, key := range candidateKeys {
+		_, err := w.getKey(key)
+		if err != nil {
+			if errors.Is(err, ErrKeyNotFound) {
+				continue
+			}
+			return -1, fmt.Errorf("failed from getKey: %w", err)
+		}
+		// Currently chooses the first usable key, but could use different logic (eg, priorities)
+		return i, nil
+	}
+
+	return -1, ErrKeyNotFound
+}
+
+// ConvertToEncryptionKey returns the Curve25519 conversion of a given Ed25519 key
+// If the key being converted is present in the wallet, then the wallet also converts and persists the matching keypair.
+func (w *BaseWallet) ConvertToEncryptionKey(key string) (string, error) {
+	signPub := base58.Decode(key)
+	encPub, err := publicEd25519toCurve25519(signPub)
+	if err != nil {
+		return "", err
+	}
+	encPubB58 := base58.Encode(encPub)
+
+	// Should we check if present in store first, or just convert and persist? Checking store likely slow
+	// checking store also introduces a variable timing dependent on secrets
+
+	kp, err := w.getKey(key)
+	if err != nil {
+		if errors.Is(ErrKeyNotFound, err) {
+			// introduces variable timing dependent on secret being present in store
+			return encPubB58, nil
```

Comment:
return "", nil

Reply:
I decided to remove that edge case from this function - there's another function to handle converting public keys that aren't expected to be in the wallet.